lib/src/components/NxGlobalSidebar/__tests__/NxGlobalSidebarFooter.test.tsx

```diff
@@ -4,63 +4,61 @@
+      quickRender = rtlRender<Props>(NxGlobalSidebarFooter, minimalProps),
+      renderEl = rtlRenderElement<Props>(NxGlobalSidebarFooter, minimalProps);
+
+  it('passes additional classes specified as props to <footer>', function() {
+    const el = renderEl({ className: 'test-class' }),
+        defaultEl = renderEl()!;
+
+    expect(el).toHaveClass('test-class');
 
-  it('renders an <footer> with nx-global-sidebar__footer class and those passed as props', function() {
-    expect(getShallowComponent({ className: 'test-class' }))
-        .toMatchSelector('footer.nx-global-sidebar__footer.test-class');
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
   it('passes additional attributes to <footer>', function() {
-    expect(getShallowComponent({ id: 'test-id' })).toHaveProp('id', 'test-id');
-    expect(getShallowComponent({ lang: 'en-ca' })).toHaveProp('lang', 'en-ca');
+    expect(renderEl({ id: 'test-id' })).toHaveAttribute('id', 'test-id');
+    expect(renderEl({ lang: 'en-ca' })).toHaveAttribute('lang', 'en-ca');
   });
 
   it('renders the Support div with a link with the passed href and text and does not render when empty', function() {
-    const supportEmpty = getShallowComponent().find('div.nx-global-sidebar__support'),
-        support = getShallowComponent({ supportText: 'Support for RSC', supportLink: '#supporturl' })
-            .find('div.nx-global-sidebar__support'),
-        supportLink = support.find(NxTextLink);
+    const { container: viewWithoutSupportText} = quickRender(),
+        viewWithSupportText = quickRender({ supportText: 'Support for RSC', supportLink: '#supporturl' }),
+        supportLink = viewWithSupportText.getByRole('link');
 
-    expect(supportEmpty).not.toExist();
-    expect(support).toContainMatchingElement('span.nx-global-sidebar__support-text');
-    expect(support.find('span.nx-global-sidebar__support-text')).toHaveText('Support for RSC');
-    expect(supportLink).toHaveProp('href', '#supporturl');
+    expect(viewWithSupportText.getByText('Support for RSC')).toBeVisible();
+    expect(supportLink).toHaveAttribute('href', '#supporturl');
+
+    expect(viewWithoutSupportText.querySelector('.nx-global-sidebar__support')).not.toBeInTheDocument();
   });
 
   it('renders the Release div with a text and version number and does not render when empty', function() {
-    const releaseEmpty = getShallowComponent().find('div.nx-global-sidebar__release'),
-        release = getShallowComponent({ releaseText: 'React Shared Components: 3.1.4' })
-            .find('div.nx-global-sidebar__release');
+    const { container: viewWithoutReleaseText} = quickRender(),
+        viewWithReleaseText = quickRender({ releaseText: 'React Shared Components: 3.1.4' });
+
+    expect(viewWithReleaseText.getByText('React Shared Components: 3.1.4')).toBeVisible();
```

Comment:
```suggestion
    expect(viewWithReleaseText.getByText('React Shared Components: 3.1.4')).toBeInTheDocument();
```

And for the other `toBeVisible` assertions

Reply:
> And for the other `toBeVisible` assertions

I went with `toBeVisible` since it already asserts that the element is in the document. Furthermore, it also checks to see if the text is actually visible to the users. https://github.com/testing-library/jest-dom#tobevisible

Let me know what you think.

Reply:
> Yep, that makes sense since it is asserting it's in the document. It may be splitting hairs a bit, but do we need to assert the text is visible? (aka is there a situation where the content is in the document but not visible?). As I said, it's probably unnecessary. I'm good with toBeVisible based on your suggestion!

I don't have a strong opinion on this, but consider that assertions of visibility are of limited effectiveness when our stylesheets aren't loaded (and the browser engine doesn't actually do layout anyway)